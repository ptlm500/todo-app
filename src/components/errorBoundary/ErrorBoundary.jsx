import React from 'react';
import PropTypes from 'prop-types';

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hasError: false,
      error: null
    };
  }

  static getDerivedStateFromError(error) {
    return {
      hasError: true,
      error
    };
  }

  render() {
    if (this.state.hasError) {
      return <h1>Uh oh! Something went wrong.</h1>;
    }

    return this.props.children;
  }
}

ErrorBoundary.propTypes = {
  children: PropTypes.node
};

export default ErrorBoundary;
