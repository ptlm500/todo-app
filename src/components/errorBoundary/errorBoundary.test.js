import React from 'react';
import renderer from 'react-test-renderer';
import ErrorBoundary from './ErrorBoundary';

function GoodComponent() {
  return (<div>{'I\'m a good component'}</div>);
}

function BadComponent() {
  throw new Error('test error');
}

describe('Error Boundary', () => {
  it('renders a child component that doesn\'t throw an error', () => {
    const tree = renderer.create(
      <ErrorBoundary>
        <GoodComponent />
      </ErrorBoundary>
    );

    expect(tree).toMatchSnapshot();
  });

  it('renders the error message when a child component throws an error', () => {
    const tree = renderer.create(
      <ErrorBoundary>
        <BadComponent />
      </ErrorBoundary>
    );

    expect(tree).toMatchSnapshot();
  });
});
