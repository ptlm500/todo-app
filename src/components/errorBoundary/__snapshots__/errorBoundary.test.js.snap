// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Boundary renders a child component that doesn't throw an error 1`] = `
<div>
  I'm a good component
</div>
`;

exports[`Error Boundary renders the error message when a child component throws an error 1`] = `
<h1>
  Uh oh! Something went wrong.
</h1>
`;
